File: qr-auth-service/src/main/java/com/app/authentiScan/payload/dto/search/QrValidationSearchDto.java
package com.app.authentiScan.payload.dto.search;

import com.app.authentiScan.enums.*;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class QrValidationSearchDto{
    private String uid;
    private ValidationMethodsEnum validationMethodsEnum;
    private ValidationStatusEnum validationStatusEnum;
    private String mobileNumber;
    private LocalDateTime validationDate;
    private WeekEnum weekEnum;
    private String month;
    private QuarterEnum quarterEnum;
    private Integer year;
    private String searchTerm;
    private ChartTypeEnum chartTypeEnum;
    private DashboardTypeEnum dashboardTypeEnum;
}

File: qr-auth-frontend/src/app/_model/dto/search/qr-validation-search-dto.ts
import { ValidationMethodsEnum } from '../../enums/validation-methods-enum';
import {ValidationStatusEnum} from '../../enums/validation-status-enum';
import {WeekEnum} from '../../enums/week-enum';
import {QuarterEnum} from '../../enums/quarter-enum';

export class QrValidationSearchDto {
  uid!: string;
  validationMethodsEnum!: string;
  validationStatusEnum!: string;
  mobileNumber!: string;
  validationDate!: string;
  weekEnum!: string;
  month!: string;
  quarterEnum!: string;
  year!: number;
  searchTerm!: string;
  chartTypeEnum!: string;
  dashboardTypeEnum!: string;

  public constructor(o?: Partial<QrValidationSearchDto>) {
    Object.assign(this, o);
  }
}
